core_chat_presentation/src/commonMain/kotlin/net/thechance/mena/core_chat/presentation/screen/shareAyaScreen/ShareAyahViewModel.kt

```diff
@@ -0,0 +1,170 @@
+            onError = { onDataLoadError() }
+        )
+    }
+
+    private fun loadContactsOperation(): Flow<PagingData<ContactUiState>> {
+        return createPagingFlow(
+            pagingSourceFactory = { createContactsPagingSource() },
+            mapper = Contact::toUi
+        )
+    }
+
+    private fun onLoadContactsSuccess(pagingData: PagingData<ContactUiState>?) {
+        updateState {
+            it.copy(contacts = flowOf(pagingData?.filter { it.isMenaUser } ?: PagingData.empty()))
+        }
+    }
+
+    private fun createContactsPagingSource(onError: ((ChatException) -> Unit)? = { onDataLoadError() })
+            : PagingSource<Int, Contact> {
+        return BasePagingSource(
+            onError = onError,
+            fetchItems = { page ->
+                if (state.value.searchQuery.isEmpty())
+                    contactsRepository.getUserContacts(page)
+                else
+                    contactsRepository.getUserContacts(page)
+                //TODO: CALL contactsRepository.getUserContactsByName(page, state.value.searchQuery)
+            }
+        )
+    }
+
+    private fun onDataLoadError() {
+        showSnackBar(
+            titleStringResource = Res.string.something_went_wrong,
+            messageStringResource = Res.string.could_not_load_the_contacts,
+            isError = true
+        )
+    }
+
+    private fun showSnackBar(
+        titleStringResource: StringResource,
+        messageStringResource: StringResource,
+        isError: Boolean = false
+    ) {
+        emitEffect(
+            ShareAyahEffect.ShowSnackBar(
+                SnackBarData(
+                    title = UiText.StringRes(titleStringResource),
+                    message = UiText.StringRes(messageStringResource),
+                    isError = isError
+                )
+            )
+        )
+    }
+
+    override fun onChangeSearchQuery(query: String) {
+        viewModelScope.launch {
+            updateState { it.copy(searchQuery = query) }
+            searchQueryFlow.value = query
+        }
+    }
+
+    override fun onClickContact(contactId: Uuid?) {
+        if (contactId == null)
+            onContactClickError()
+        else
+            tryToExecute(
+                execute = { chatRepository.getChatByContactUserId(userId = contactId) },
+                onSuccess = { chat ->
+                    onContactClickSuccess(
+                        chatId = chat.id,
+                        chatName = chat.name
+                    )
+                },
+                onError = { onContactClickError() }
+            )
+    }
+
+    override fun onClickBack() {
+        emitEffect(effect = ShareAyahEffect.NavigateBack)
+    }
+
+    override fun onClickClearQuery() {
+        updateContacts("")
+        searchQueryFlow.value = ""
+    }
+
+    private fun updateContacts(searchQuery: String) {
+        updateState { it.copy(searchQuery = "") }
+        getContacts()
+    }
```

Comment:
i think there is no need to create extra function for this case, that is my opinion 
```suggestion
    override fun onClickClearQuery() {
        updateState { it.copy(searchQuery = "") }
        searchQueryFlow.value = ""
                getContacts()
    }
```

Reply:
I made update contacts just for reusability as it might be called with any query rather than updating search query and get contacts each time